insonmnia/oracle/oracle.go

```diff
@@ -0,0 +1,179 @@
+package oracle
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/noxiouz/zapctx/ctxlog"
+	"github.com/sonm-io/core/blockchain"
+	"github.com/sonm-io/core/util"
+	"go.uber.org/zap"
+)
+
+type Oracle struct {
+	key *ecdsa.PrivateKey
+	cfg *Config
+
+	logger *zap.Logger
+
+	bch blockchain.API
+
+	actualPrice  *big.Int
+	currentPrice *big.Int
+}
+
+func NewOracle(ctx context.Context, key *ecdsa.PrivateKey, cfg *Config) (*Oracle, error) {
+	logger := ctxlog.GetLogger(ctx)
+
+	var mode string
+	if cfg.Oracle.Mode {
+		mode = "master"
+	} else {
+		mode = "slave"
+	}
+	logger.Info("start USD-SNM Oracle",
```

Comment:
But you're creating an instance, not starting it. I think this log line should be into the `.Serve()` method.